SIPS/sip-29.md

```diff
@@ -35,15 +35,52 @@ in uppercase in this document are to be interpreted as described in [RFC
 2. Any time an asset needs to be identified, it MUST use the [CAIP-19][caip-19]
 representation.
 
-### Snap Assets API
+### Snap Manifest
+
+This SIP introduces a new permission named `endowment:assets`.
+This permission grants a snap the ability to provide asset information to the client.
+
+This permission is specified as follows in `snap.manifest.json` files:
+
+```json
+{
+  "initialPermissions": {
+    "endowment:assets": {}
+  }
+}
+```
+
+### Snap Implementation
 
 Two methods are defined in the Snap Assets API:
 
+Any snap that wishes to provide asset information **MUST** implement the following API:
+
 #### Get Token Description
 
+```typescript
+import { OnAssetDescriptionHandler } from "@metamask/snap-types";
+
+export const onAssetDescription: OnAssetDescriptionHandler = async ({
```

Comment:
Wondering if we need the split to have two different handlers 🤔 

Perhaps a question for the original SIP, but wondering if we could simply have a lookup function that would return the known pricing and metadata combined reducing the amount of potential calls

Reply:
E.g. `onAssetLookup` or something

Reply:
@danroc what's your take on this ?

Reply:
We'll need to poll frequently for the asset rate on the client side and the asset description might be unused data in this case. If the snap doesn't cache the asset metadata it will mean that a call to an API is done everytime for nothing.

Reply:
That's a good point 🤔

Reply:
Indeed, I agree with @GuillaumeRx, those are two distinct concerns and the methods are not called with same frequency.